boards/arm/nrf5340dk_nrf5340/nrf5340_cpunet_reset.c

```diff
@@ -83,7 +83,8 @@ static int remoteproc_mgr_boot(const struct device *dev)
 	 * this case do the remainder of actions to properly configure and
 	 * boot the Network MCU.
 	 */
-#if defined(SHM_BASE_ADDRESS) && (SHM_BASE_ADDRESS != 0)
+#if !defined(CONFIG_BOARD_KEEP_SHM) && \
```

Comment:
OpenAMP takes care of initializing memory during start-up sequence using `virtio_set_status()`. If shared memory was cleared here, depending on initialization sequence, data already set by OpenAMP were cleared here.

When I disable clearing shared memory, OpenAMP works correctly regardless initialization sequence.

Reply:
We've discussed this part offline with @kapi-no. It seems that the code clearing shred memory is not needed anymore, because it was intended to be used with OpenAMP. @ioannisg, could you confirm if it would be OK to remove `memset()` from this function?

Reply:
Discussed offline - the conclusion is to remove the initialization of the shared memory to zero and the CONFIG_BOARD_KEEP_SHM configuration. The libraries, which use the shared memory (e.g. OpenAMP), should take care of initialization if necessary.